limits/aio/storage/memcached.py

```diff
@@ -150,3 +240,115 @@ async def check(self) -> bool:
 
     async def reset(self) -> Optional[int]:
         raise NotImplementedError
+
+    def _current_window_key(
```

Comment:
Perhaps these could be extracted out to be reused by any storage (async or not) that uses timestamp based sliding window keys? Optionally it could just be one function since they are never needed exclusively. If this is a class then we could even detect such storages (for example in tests) by using `isinstance`.

For example:

```python
class TimestampedSlidingWindow:
    @classmethod 
    def sliding_window_keys(cls, key: str, expiry: int, at: float) -> Tuple[str, str]:
        ....
```

Reply:
Yes, this is much cleaner now! I've amended the first commit.

Reply:
Thank you!